packages/components/nodes/documentloaders/Folder/Folder.ts

```diff
@@ -54,14 +61,17 @@ class Folder_DocumentLoaders implements INode {
         const textSplitter = nodeData.inputs?.textSplitter as TextSplitter
         const folderPath = nodeData.inputs?.folderPath as string
         const metadata = nodeData.inputs?.metadata
+        const additionalLoaders = nodeData.inputs?.additionalLoaders
+        const parsedLoaders = additionalLoaders ? ( typeof metadata === 'object' ? additionalLoaders: JSON.parse( additionalLoaders ) ) : []
 
         const loader = new DirectoryLoader(folderPath, {
             '.json': (path) => new JSONLoader(path),
             '.txt': (path) => new TextLoader(path),
             '.csv': (path) => new CSVLoader(path),
             '.docx': (path) => new DocxLoader(path),
             // @ts-ignore
-            '.pdf': (path) => new PDFLoader(path, { pdfjs: () => import('pdf-parse/lib/pdf.js/v1.10.100/build/pdf.js') })
+            '.pdf': (path) => new PDFLoader(path, { pdfjs: () => import('pdf-parse/lib/pdf.js/v1.10.100/build/pdf.js') }),
+            ...parsedLoaders
```

Comment:
I don't think it will work. Because you need specific loader for specific file types, otherwise if let's say you have a file with extension `.ppt`, there is no loader to process the file

Reply:
Well this is the tricky part because you have to leave the loader selection in the hands of the user right? 

This might require an addition of a new type of user control to flowise-ui that allows a user to type in a file type and select a loader from the list. Then in this code you could create a new function that returns a instance of the loader depending on the selected loader. 

There are hundreds of file types that don't fit in this list that could use just the TextLoader. Alternatively do you suggest that I add some of the basic code extensions here? like .cs, .php, .js, .xml, .cpp, etc?

Reply:
I think for now we can just add the other extensions here. For example: 

`'.cs': (path) => new TextLoader(path),`

You can refer to all available code extensions [here](https://github.com/FlowiseAI/Flowise/blob/main/packages/components/nodes/textsplitters/CodeTextSplitter/CodeTextSplitter.ts#L33)